coderd/workspaceapps/apptest/apptest.go

```diff
@@ -1549,3 +1564,31 @@ func testReconnectingPTY(ctx context.Context, t *testing.T, client *codersdk.Cli
 	// Ensure the connection closes.
 	require.ErrorIs(t, tr.ReadUntil(ctx, nil), io.EOF)
 }
+
+// Accessing an app should update the workspace's LastUsedAt.
+// NOTE: Despite our efforts with the flush channel, this is inherently racy.
+func assertWorkspaceLastUsedAtUpdated(t testing.TB, details *Details) {
+	t.Helper()
+
+	<-time.After(testutil.IntervalMedium) // Wait for Bicopy to finish.
+	details.FlushStats()
+	// Wait for stats to fully flush.
+	require.Eventually(t, func() bool {
+		ws, err := details.SDKClient.Workspace(context.Background(), details.Workspace.ID)
+		assert.NoError(t, err)
+		return ws.LastUsedAt.After(details.Workspace.LastUsedAt)
+	}, testutil.WaitShort, testutil.IntervalMedium, "workspace LastUsedAt not updated when it should have been")
+}
+
+// Except when it sometimes shouldn't (e.g. no access)
+// NOTE: Despite our efforts with the flush channel, this is inherently racy.
+func assertWorkspaceLastUsedAtNotUpdated(t testing.TB, details *Details) {
+	t.Helper()
+
+	<-time.After(testutil.IntervalMedium) // Wait for Bicopy to finish.
+	details.FlushStats()
+	<-time.After(testutil.IntervalMedium) // Wait for stats to fully flush.
+	ws, err := details.SDKClient.Workspace(context.Background(), details.Workspace.ID)
+	require.NoError(t, err)
+	require.Equal(t, ws.LastUsedAt, details.Workspace.LastUsedAt, "workspace LastUsedAt updated when it should not have been")
+}
```

Comment:
review: this is where all the race gremlins live:
- Bicopy does not appear to exit immediately so if we flush stats immediately we may not get a finished session.
- Even though a completed flush _should_ mean that LastUsedAt has been bumped, I've observed some wiggle room (especially when wsproxy is involved).

Reply:
Why don't we try looping the flush until we either time out or have the stats we're looking for? Might make sense to have `FlushStats` take a context too, just to be sure.